### ISSUE TITLE
Warnings by AllYourBase tool

### ISSUE THREAD

**Issue by Alicecomma** [CombatExtendedRWMod/CombatExtended#1041]
```
BaseBullet in file C:\Games\RimWorld2282Win64\Mods\CombatExtended\Defs\Ammo\AmmoBases.xml
MoteBase in file C:\Games\RimWorld2282Win64\Mods\CombatExtended\Defs\Effects\Mote_Visual.xml
TribalBase in file C:\Games\RimWorld2282Win64\Mods\CombatExtended\Defs\PawnKindDefs\PawnKinds_CE.xml
MercenaryBase in file C:\Games\RimWorld2282Win64\Mods\CombatExtended\Defs\PawnKindDefs\PawnKinds_CE.xml
BuildingBase in file C:\Games\RimWorld2282Win64\Mods\CombatExtended\Defs\ThingDefs_Buildings\Buildings_Defenses.xml
BenchBase in file C:\Games\RimWorld2282Win64\Mods\CombatExtended\Defs\ThingDefs_Buildings\Buildings_Production.xml
UnfinishedBase in file C:\Games\RimWorld2282Win64\Mods\CombatExtended\Defs\ThingDefs_Items\Items_Unfinished.xml
ApparelBase in file C:\Games\RimWorld2282Win64\Mods\CombatExtended\Defs\ThingDefs_Misc\Apparel_Various.xml
BaseFilth in file C:\Games\RimWorld2282Win64\Mods\CombatExtended\Defs\ThingDefs_Misc\Filth_Various.xml
BaseGrenadeProjectile in file C:\Games\RimWorld2282Win64\Mods\CombatExtended\Defs\ThingDefs_Misc\Weapons_Grenades.xml
BaseProjectileNeolithic in file C:\Games\RimWorld2282Win64\Mods\CombatExtended\Defs\ThingDefs_Misc\Weapons_RangedNeolithic.xml
BaseWeaponNeolithic in file C:\Games\RimWorld2282Win64\Mods\CombatExtended\Defs\ThingDefs_Misc\Weapons_RangedNeolithic.xml
PlantBase in file C:\Games\RimWorld2282Win64\Mods\CombatExtended\Defs\ThingDefs_Plants\Plants_Bases.xml
```

**Comment by Alicecomma** [comment]
AmmoBases.xml:
- Why exactly is BaseBullet a Class="CombatExtended.AmmoDef". There is absolutely nothing that warrants this - all projectiles should be ThingDef, not AmmoDef
- BaseBullet should be renamed to CombatExtended_BaseBullet at best
- - Practically all ammo files inherit from BaseBullet

Mote_Visual.xml:
- No reason for the file to be there, carbon copy of Mote_Visual.xml

PawnKinds_CE.xml:
- TribalBase is carbon copy of Core's TribalBase, should be removed
- MercenaryBase idem

Buldings_Defenses.xml:
- BuildingBase is carbon copy of Core's BuildingBase

Buildings_Production.xml:
- BenchBase is carbon copy

Items_Unfinished.xml:
- UnfinishedBase is carbon copy

Apparel_Various.xml:
- ApparelBase is carbon copy

Filth_Various.xml:
- BaseFilth is carbon copy

Weapons_Grenades.xml:
- BaseGrenadeProjectile has some things to override w.r.t. Core, so a differently named base should be used, e.g CombatExtended_BaseGrenadeProjectile. Perhaps it can override earlier in the Abstract chain.

Weapons_RangedNeolithic.xml:
- BaseWeaponNeolithic is not even used
- BaseProjectileNeolithic is used by:
- - BasePilumProjectile in Javelins.xml

Plants_Bases.xml:
- PlantBase is carbon copy

**Comment by Alicecomma** [comment]
This leaves us with:
- BaseGrenadeProjectile 
- BaseBullet being an AmmoDef and everything inheriting from it

**Comment by N7Huntsman** [comment]
In regards to the carbon-copy bases, here's the conversation I had with NIA barely a week ago:

> N7Huntsman 02/12/2020
Is there any crucial reason that CE overwrites (not patches, overwrites) vanilla "BuildingBase" with a completely identical version, or is that something I can get rid of?
N7Huntsman 02/12/2020
Ditto with "Plantbase."
NoImageAvailable 02/12/2020
It'll open up a shitload of compatibility issues with other mods that touch those base defs
N7Huntsman 02/12/2020
Granted, but isn't best practice just for all mods to leave the vanilla Bases alone?
And with Combat Extended generally being pretty high in the load order, won't subsequently-loaded mods still potentially affect the Bases anyway?
N7Huntsman 02/12/2020
Who currently owns/maintains the sheets for Melee, Guns, and Race stats?
NoImageAvailable 02/12/2020
The inheritance code is really weird and unintuitive
You'd want to either rename the bases to something unique or leave them as is
And renaming them is liable to break compatibility with mods targeting vanilla bases
So it's best to not try to fix what ain't broke

**Comment by Alicecomma** [comment]
Some of these bases are literal 1:1 copies for highly non-critical defs, such as motes, plants, filth, unfinished items, crafting benches. Thus, compatibility with mods modifying those can be broken, even if CE does absolutely nothing of interest with those bases.

And then, some of the bases which are useful (ammo), they are misconfigured. There is no reason for projectiles to have an ammoDef class, I don't understand why it is introduced at all.

**Comment by N7Huntsman** [comment]
I can only speak to the duplicate bases, wasn't aware of any misconfigurations. I agree the duplicate bases should be done away with if possible, I'm just saying that we're moving a lot of new stuff into CE at present, so let's make sure that stuff is stable before we start pulling wires--that way, is stuff starts breaking, we know why.

**Comment by zozilin** [comment]
I agree with N7 on this - not critical, can be tested with multiple mods later. Split that Javelin fix from everything else.
